### PR TITLE
Retrieve viewLayer and version from dependencies and support @web/dev-server-storybook

### LINKED ISSUE #317: Support prebuilt Storybook package
Reported [here](https://app.intercom.com/a/apps/zj7sn9j1/inbox/inbox/all/conversations/27253915916).

Modern Web's [Dev Server Storybook](https://github.com/modernweb-dev/web/tree/master/packages/dev-server-storybook) uses a prebuilt Storybook manager UI and doesn't require installing `@storybook/web-components`. Their [guide](https://modern-web.dev/guides/dev-server/storybook/) also doesn't suggest installing it.

This is problematic because the Chromatic CLI will look for a supported view layer package and not find any. It can be resolved by installing `@storybook/web-components` manually, but we should just support this use case out of the box.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1199921881886840/1200198737996367) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by ghengeveld**
Fixes #317

This makes installing the Storybook viewlayer package to node_modules optional when using `--storybook-build-dir`. We won't need the Storybook CLI in that case, so we don't care whether it's installed.

I also added explicit support for `@web/dev-server-storybook` which required a bit of a refactor.

### CODE REVIEW

**Review by ndelangen — APPROVED** [review]

**Review by tmeasday — APPROVED** [review]
Great!

**Review by ndelangen — APPROVED** [review]

File: bin/lib/getStorybookInfo.js [hunk 1]
```diff
@@ -66,31 +66,66 @@ const timeout = (count) =>
...
+  // Pull the viewlayer from dependencies in package.json
+  const dep = Object.entries(packageJson.dependencies || {}).find(([p]) => viewLayers[p]);
+  const devDep = Object.entries(packageJson.devDependencies || {}).find(([p]) => viewLayers[p]);
+  const peerDep = Object.entries(packageJson.peerDependencies || {}).find(([p]) => viewLayers[p]);
+  const dependency = dep || devDep || peerDep;
+
+  if (!dependency) {
+    throw new Error(
+      'Could not find a supported Storybook viewlayer package in your package.json dependencies. Make sure one is installed.'
+    );
```
Comment (ndelangen) [inline bin/lib/getStorybookInfo.js:95]: Add a url to a documentation page about this?

**Review by ndelangen — APPROVED** [review]

File: bin/ui/messages/errors/noViewLayerPackage.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import chalk from 'chalk';
+import dedent from 'ts-dedent';
+
+import { error } from '../../components/icons';
+
+export default (pkg) =>
+  dedent(chalk`
+    ${error} {bold Storybook package not installed}
+    Could not find {bold ${pkg}} in {bold node_modules}.
+    Most likely you forgot to run {bold npm install} or {bold yarn} before running Chromatic.
```
Comment (domyen) [inline bin/ui/messages/errors/noViewLayerPackage.js:10]:
```suggestion
    Most likely, you forgot to run {bold npm install} or {bold yarn} before running Chromatic.
```

**Comment by domyen** [comment]
Nice! This resolves #324 which has been reported by customers lately.

**Pull request merged**